packages/salesforcedx-vscode-core/src/commands/forceSourceDeploySourcePath.ts

```diff
@@ -152,7 +84,9 @@ export async function forceSourceDeploySourcePath(sourceUri: vscode.Uri) {
   const commandlet = new SfdxCommandlet(
     new SfdxWorkspaceChecker(),
     new FilePathGatherer(sourceUri),
-    new ForceSourceDeploySourcePathExecutor(),
+    useBetaRetrieve(sourceUri)
+      ? new LibraryDeploySourcePathExecutor()
+      : new ForceSourceDeploySourcePathExecutor(),
```

Comment:
Because deploy on save uses the "multiple source path" command, it can't take advantage of this logic at the moment if we're only deploying one at a time.

Reply:
True, I'm not sure if the "old" version worked with deploy on save, I think it did but let me verify